internal/lib/stats/stats_server_linux.go

```diff
@@ -40,11 +47,12 @@ func (ss *StatsServer) updateSandbox(sb *sandbox.Sandbox) *types.PodSandboxStats
 		if c.StateNoLock().Status == oci.ContainerStateStopped {
 			continue
 		}
-		cStats, err := ss.Runtime().ContainerStats(context.TODO(), c, sb.CgroupParent())
+		cgstats, err := ss.Runtime().ContainerStats(context.TODO(), c, sb.CgroupParent())
```

Comment:
There is a problem here: you're expecting `ContainerStats()` to return a `CgroupStats` structure, but you're modifying `ContainerStats` only in the next commit.
Your first commit then doesn't build - you get an error when you try to use `cgstats` as a parameter to `containerCRIStats()`

```
internal/lib/stats/stats_server_linux.go:55:31: cannot use cgstats (variable of type *"k8s.io/cri-api/pkg/apis/runtime/v1".ContainerStats) as *cgmgr.CgroupStats value in argument to containerCRIStats
```

If you don't want to merge the two commits (which would make it pretty big), maybe you could introduce that function code changes only in the second commit?
Not sure what option is best, but I think we need to have independently buildable commits

Reply:
I would go with merging the two commits if no problem with that. it will be a big commit but I guess all changes are relevant. ( will make the commit message a bit meaningful)